R/pretty_lollipop_plot.R

```diff
@@ -120,98 +217,453 @@ pretty_lollipop_plot <- function(
+                show.legend = FALSE
+            ) +
+            geom_text(
+                data = domain_data, 
+                aes(
+                    x = text.position, 
+                    y = 0,
+                    angle = 90, 
+                    label = paste0(
+                        text.label,
+                        "\n",
+                        domain_data$p_value
+                        )
                 )
-        ) +
-        geom_point(
-            data = gene_counts, 
-            aes(
-                x = AA, 
-                y = mutation_count, 
-                color = Variant_Classification, 
-                size = mutation_count
+            ) +
+            theme_void() +
+            theme(axis.text.x = element_text(vjust = rel(0.5)))
+
+        # Function for mutation lollipop plot
+        mutation_plot <- function(gene_counts){
+            ggplot() + 
+            geom_segment(
+                data = gene_counts, 
+                aes(
+                    x = AA, 
+                    xend = AA, 
+                    y = 0, 
+                    yend = mutation_count
+                )
+            ) +
+            geom_point(
+                data = gene_counts, 
+                aes(
+                    x = AA, 
+                    y = mutation_count, 
+                    color = Variant_Classification, 
+                    size = mutation_count
                 )
-        ) +
-        # Background rectangle for regions without domain data
-        geom_rect(
-            aes(
-                xmin = x_min, 
-                xmax = x_max, 
-                ymin = -0.2, 
-                ymax = 0.2
-                ), 
-            fill = "black", 
-            color = "black"
-        ) + 
-        # Domain rectangles
-        geom_rect(
-            data = domain_data, 
-            aes(
-                xmin = start.points, 
-                xmax = end.points, 
-                ymin = -0.4, 
-                ymax = 0.4, 
-                fill = color
-                ), 
-            color = "black", 
-            show.legend = FALSE
-        ) +
-        geom_text(
-            data = domain_data, 
-            aes(
-                x = text.position, 
-                y = 0, 
-                label = text.label
+            ) + 
+            ggpubr::theme_pubr() 
+        }
+
+        # common legend
+        co_legend_data <- mutation_plot(gene_counts) +
+            scale_size_continuous(
+                breaks = seq(
+                    floor(min(gene_counts$mutation_count)), 
+                    ceiling(max(gene_counts$mutation_count)), 
+                    by = 1
+                ),
+                labels = function(x) format(x, scientific = FALSE)  
+            )  +
+            scale_color_manual(name = "Mutation Type", values = colours_manual) +
+            theme(legend.position = "right")
+        gg_co_legend <- as_ggplot(get_legend(co_legend_data))
```

Comment:
This errors as it is currently, I think it needs to specify `ggpubr::get_legend`